ast.go

```diff
@@ -67,6 +67,42 @@ type Writer interface {
 	WriteStr(node StrVal) error
 }
 
+// WriterWrapper defines an interface for writing a AST as SQL.
```

Comment:
What is `WriterWrapper` for?

Reply:
Oops, it was an artifact of testing. You can pass in any `io.Writer` to it and it will produce a `squalor.Writer`.

I can put it in a separate PR as it is useful but not part of this PR.